host/esutils/client_os2.go

```diff
@@ -0,0 +1,162 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package esutils
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	os2Client struct {
+		client *opensearch.Client
+	}
+)
+
+func newOS2Client(url string) (*os2Client, error) {
+
+	esClient, err := opensearch.NewClient(
+		opensearch.Config{
+			Addresses:    []string{url},
+			MaxRetries:   5,
+			RetryBackoff: func(i int) time.Duration { return time.Duration(i) * 100 * time.Millisecond },
+		},
+	)
+
+	return &os2Client{
+		client: esClient,
+	}, err
+}
+
+func (os2 *os2Client) PutIndexTemplate(t *testing.T, templateConfigFile, templateName string) {
+	// This function is used exclusively in tests. Excluding it from security checks.
+	// #nosec
+	template, err := os.Open(templateConfigFile)
+	require.NoError(t, err)
+
+	req := opensearchapi.IndicesPutIndexTemplateRequest{
+		Body: template,
+		Name: templateName,
+	}
+
+	resp, err := req.Do(createContext(), os2.client)
+	require.NoError(t, err)
+	require.False(t, resp.IsError(), fmt.Sprintf("OS2 error: %s", resp.String()))
+	resp.Body.Close()
+
+}
+
+func (os2 *os2Client) CreateIndex(t *testing.T, indexName string) {
+	existsReq := opensearchapi.IndicesExistsRequest{
+		Index: []string{indexName},
+	}
+	resp, err := existsReq.Do(createContext(), os2.client)
+	require.NoError(t, err)
+
+	if resp.StatusCode == http.StatusOK {
+		deleteReq := opensearchapi.IndicesDeleteRequest{
+			Index: []string{indexName},
+		}
+		resp, err := deleteReq.Do(createContext(), os2.client)
+		require.Nil(t, err)
+		require.False(t, resp.IsError(), fmt.Sprintf("OS2 error: %s", resp.String()))
+	}
+
+	resp.Body.Close()
+
+	createReq := opensearchapi.IndicesCreateRequest{
+		Index: indexName,
+	}
+
+	resp, err = createReq.Do(createContext(), os2.client)
+	require.NoError(t, err)
+	require.False(t, resp.IsError(), fmt.Sprintf("OS2 error: %s", resp.String()))
+	resp.Body.Close()
+}
+
+func (os2 *os2Client) DeleteIndex(t *testing.T, indexName string) {
+	deleteReq := opensearchapi.IndicesDeleteRequest{
+		Index: []string{indexName},
+	}
+	resp, err := deleteReq.Do(createContext(), os2.client)
+	require.NoError(t, err)
+	require.False(t, resp.IsError(), fmt.Sprintf("OS2 error: %s", resp.String()))
+	resp.Body.Close()
+
+}
+
+func (os2 *os2Client) PutMaxResultWindow(indexName string, maxResultWindow int) error {
```

Comment:
while you're in here: these also seem to only be used in tests.  add a `*testing.T` arg maybe, to make that explicit?

(no requirement, it's already improved and this is pretty easy to verify)

Reply:
done